### PR TITLE
ci: enable `strictInputTypes`

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]

File: projects/core/abstract/abstract-hint-options.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import {TuiDirection, TuiHintModeT} from '@taiga-ui/core/types';
+import {TuiBrightness, TuiDirection, TuiHintModeT} from '@taiga-ui/core/types';
 
 export interface TuiAbstractHintOptions {
-    readonly mode: TuiHintModeT | null;
+    readonly mode: TuiHintModeT | TuiBrightness | null;
```
Comment (waterplea) [inline projects/core/abstract/abstract-hint-options.ts:4]: We do not support whole `TuiBrightness`, only `onDark` which is already in `TuiHintModeT`.

File: projects/core/components/dialog/dialog.directive.ts [hunk 2]
```diff
@@ -15,6 +15,6 @@ import {TuiDialogService} from './dialog.service';
     inputs: ['options: tuiDialogOptions', 'open: tuiDialog'],
     outputs: ['openChange: tuiDialogChange'],
 })
-export class TuiDialogDirective<T> extends AbstractTuiDialogDirective<
-    TuiDialogOptions<T>
+export class TuiDialogDirective extends AbstractTuiDialogDirective<
```
Comment (waterplea) [inline projects/core/components/dialog/dialog.directive.ts:18]: Why this change?

File: projects/core/components/hints-host/hint-box/hint-box.component.ts [hunk 3]
```diff
@@ -83,7 +83,7 @@ export class TuiHintBoxComponent {
     }
 
     @HostBinding('attr.data-mode')
-    get mode(): TuiHintModeT | null {
+    get mode(): TuiHintModeT | TuiBrightness | null {
```
Comment (waterplea) [inline projects/core/components/hints-host/hint-box/hint-box.component.ts:86]: Same here

File: projects/demo/src/modules/charts/axes/axes.component.ts [hunk 4]
```diff
@@ -47,11 +47,11 @@ export class ExampleTuiAxesComponent {
         'hidden',
     ];
 
-    readonly labelsVariants = [
+    readonly labelsVariants: readonly (readonly string[])[] = [
         [],
         ['', '25%', '50%', '100%'],
         ['One', 'Two', 'Three'],
-        ['One', null, '', 'Two and a half', 'Three', null],
```
Comment (waterplea) [inline projects/demo/src/modules/charts/axes/axes.component.ts:54]: Are you sure `null` and `''` are the same? There specifically were both cases in the array here.

File: projects/demo/src/modules/charts/line-chart/examples/1/index.ts [hunk 5]
```diff
@@ -11,7 +12,7 @@ import {TUI_DEFAULT_STRINGIFY} from '@taiga-ui/cdk';
     encapsulation,
 })
 export class TuiLineChartExample1 {
-    readonly value = [
+    readonly value: ReadonlyArray<TuiPoint> = [
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-chart/examples/1/index.ts:15]: `readonly TuiPoint[]` is a more modern way to write this :) Let's use it. I tend to use `ReadonlyArray` only for two dimensional arrays, not a fan of `readonly (readonly string[])[]` — hard to read imho

File: projects/demo/src/modules/charts/line-chart/examples/2/index.ts [hunk 6]
```diff
@@ -10,7 +11,7 @@ import {encapsulation} from '@demo/emulate/encapsulation';
     encapsulation,
 })
 export class TuiLineChartExample2 {
-    readonly value = [
+    readonly value: ReadonlyArray<TuiPoint> = [
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-chart/examples/2/index.ts:14]: Same here.

File: projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts [hunk 7]
```diff
@@ -65,7 +65,9 @@ export class ExampleTuiLineDaysChartComponent {
         ],
     ];
 
-    value = this.valueVariants[0];
+    value: ReadonlyArray<[TuiDay, number]> = this.valueVariants[0] as ReadonlyArray<
+        [TuiDay, number]
```
Comment (waterplea) [inline projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts:69]: No need to typecast if you type `valueVariants`.

File: projects/demo/src/modules/components/accordion/accordion.component.ts [hunk 8]
```diff
@@ -63,9 +63,9 @@ export class ExampleTuiAccordionComponent {
         LESS: import('!!raw-loader!./examples/4/index.less'),
     };
 
-    readonly bordersVariants = ['all', 'top-bottom'];
+    readonly bordersVariants: ('all' | 'top-bottom')[] = ['all', 'top-bottom'];
```
Comment (waterplea) [inline projects/demo/src/modules/components/accordion/accordion.component.ts:66]: `as const`?

File: projects/demo/src/modules/components/carousel/examples/4/index.html [hunk 9]
```diff
@@ -7,7 +7,7 @@
 </ng-template>
 
 <ng-template
-    [tuiDialogOptions]="{ size: 's', header: header }"
+    [tuiDialogOptions]="$any({ size: 's', header: header })"
```
Comment (waterplea) [inline projects/demo/src/modules/components/carousel/examples/4/index.html:10]: I saw this issue and I'd like to fix it instead of adding `$any`. Let's dig into it.

File: projects/demo/src/modules/components/combo-box/combo-box.component.ts [hunk 10]
```diff
@@ -103,17 +106,16 @@ export class ExampleTuiComboBoxComponent extends AbstractExampleTuiControl {
...
         TUI_STRICT_MATCHER,
         (item, search, stringify) =>
             Number.parseInt(stringify(item).match(/\d+/g)![0], 10) ===
             Number.parseInt(search, 10),
         null,
     ];
 
-    strictMatcher = this.strictMatcherVariants[0];
+    strictMatcher: TuiStringMatcher<Account> | null = this
+        .strictMatcherVariants[0] as TuiStringMatcher<Account> | null;
```
Comment (waterplea) [inline projects/demo/src/modules/components/combo-box/combo-box.component.ts:118]: Unnecessary typecast

File: projects/demo/src/modules/components/dialog/examples/6/index.html [hunk 11]
```diff
@@ -1,7 +1,7 @@
 <button tuiButton type="button" size="m" (click)="showDialog()">Show</button>
 <ng-template
     let-observer
-    [tuiDialogOptions]="{ label: 'Declarative directive', size: 's' }"
+    [tuiDialogOptions]="$any({ label: 'Declarative directive', size: 's' })"
```
Comment (waterplea) [inline projects/demo/src/modules/components/dialog/examples/6/index.html:4]: Same here

File: projects/demo/src/modules/components/input-date-time/input-date-time.component.ts [hunk 12]
```diff
@@ -68,7 +68,7 @@ export class ExampleTuiInputDateTimeComponent extends AbstractExampleTuiControl
         [this.today.append({day: -1}), new TuiTime(12, 20)],
     ];
 
-    min = this.minVariants[0];
+    min: TuiDay = this.minVariants[0] as TuiDay;
```
Comment (waterplea) [inline projects/demo/src/modules/components/input-date-time/input-date-time.component.ts:71]: Add type to `minVariants` instead

File: projects/demo/src/modules/components/input-date-time/input-date-time.component.ts [hunk 13]
```diff
@@ -78,7 +78,7 @@ export class ExampleTuiInputDateTimeComponent extends AbstractExampleTuiControl
         [this.today.append({day: +1}), new TuiTime(16, 20)],
     ];
 
-    max = this.maxVariants[0];
+    max: TuiDay = this.maxVariants[0] as TuiDay;
```
Comment (waterplea) [inline projects/demo/src/modules/components/input-date-time/input-date-time.component.ts:81]: Same here

File: projects/demo/src/modules/components/input-tag/input-tag.component.ts [hunk 14]
```diff
@@ -112,5 +114,9 @@ export class ExampleTuiInputTagComponent extends AbstractExampleTuiControl {
         item => item[0] === 'T',
     ];
 
-    disabledItemHandler = this.disabledItemHandlerVariants[0];
+    disabledItemHandler: TuiHandler<string | TuiStringifiableItem<unknown>, boolean> =
+        this.disabledItemHandlerVariants[0] as TuiHandler<
```
Comment (waterplea) [inline projects/demo/src/modules/components/input-tag/input-tag.component.ts:118]: Again, type the variants instead

File: projects/demo/src/modules/components/island/island.component.ts [hunk 15]
```diff
@@ -46,9 +46,13 @@ export class ExampleTuiIslandComponent {
...
 
-    readonly textAlignVariants = ['left', 'right', 'center'];
+    readonly textAlignVariants: ('left' | 'right' | 'center')[] = [
+        'left',
+        'right',
+        'center',
+    ];
 
-    textAlign = this.textAlignVariants[0];
+    textAlign: 'left' | 'right' | 'center' = this.textAlignVariants[0];
```
Comment (waterplea) [inline projects/demo/src/modules/components/island/island.component.ts:55]: Should probably be enough to type the variants.

File: projects/demo/src/modules/directives/swipe/examples/2/index.ts [hunk 16]
```diff
@@ -12,7 +12,7 @@ import {Subject} from 'rxjs';
     encapsulation,
 })
 export class TuiSwipeExample2 {
-    readonly open$ = new Subject();
+    readonly open$: Subject<boolean> = new Subject();
```
Comment (waterplea) [inline projects/demo/src/modules/directives/swipe/examples/2/index.ts:15]: Should be enough to type `new Subject<boolean>()`

File: projects/core/abstract/abstract-hint-options.ts (hunk 1 above)
Comment (splincode) [reply projects/core/abstract/abstract-hint-options.ts:4]: But when I change it

image

I have problems

image

File: projects/demo/src/modules/components/carousel/examples/4/index.html (hunk 9 above)
Comment (splincode) [reply projects/demo/src/modules/components/carousel/examples/4/index.html:10]: can we leave TODO here for now? and solve it in the next PR?

File: projects/demo/src/modules/components/dialog/examples/6/index.html (hunk 11 above)
Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/6/index.html:4]: can we leave TODO here for now? and solve it in the next PR?

File: projects/core/components/dialog/dialog.directive.ts (hunk 2 above)
Comment (splincode) [reply projects/core/components/dialog/dialog.directive.ts]: fixed

File: projects/demo/src/modules/charts/axes/axes.component.ts (hunk 4 above)
Comment (splincode) [reply projects/demo/src/modules/charts/axes/axes.component.ts]: fixed

File: projects/core/components/hints-host/hint-box/hint-box.component.ts (hunk 3 above)
Comment (splincode) [reply projects/core/components/hints-host/hint-box/hint-box.component.ts:86]: comment above

File: projects/demo/src/modules/charts/line-chart/examples/1/index.ts (hunk 5 above)
Comment (splincode) [reply projects/demo/src/modules/charts/line-chart/examples/1/index.ts]: fixed

File: projects/demo/src/modules/charts/line-chart/examples/2/index.ts (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/charts/line-chart/examples/2/index.ts]: fixed

File: projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts (hunk 7 above)
Comment (splincode) [reply projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts]: fixed

File: projects/demo/src/modules/components/accordion/accordion.component.ts (hunk 8 above)
Comment (splincode) [reply projects/demo/src/modules/components/accordion/accordion.component.ts]: fixed

File: projects/demo/src/modules/components/combo-box/combo-box.component.ts (hunk 10 above)
Comment (splincode) [reply projects/demo/src/modules/components/combo-box/combo-box.component.ts]: fixed

File: projects/demo/src/modules/components/input-date-time/input-date-time.component.ts (hunk 12 above)
Comment (splincode) [reply projects/demo/src/modules/components/input-date-time/input-date-time.component.ts]: fixed

File: projects/demo/src/modules/components/input-tag/input-tag.component.ts (hunk 14 above)
Comment (splincode) [reply projects/demo/src/modules/components/input-tag/input-tag.component.ts]: fixed

File: projects/demo/src/modules/components/island/island.component.ts (hunk 15 above)
Comment (splincode) [reply projects/demo/src/modules/components/island/island.component.ts]: fixed

File: projects/demo/src/modules/directives/swipe/examples/2/index.ts (hunk 16 above)
Comment (splincode) [reply projects/demo/src/modules/directives/swipe/examples/2/index.ts]: fixed

File: projects/demo/src/modules/components/input-date-time/input-date-time.component.ts (hunk 13 above)
Comment (splincode) [reply projects/demo/src/modules/components/input-date-time/input-date-time.component.ts]: fixed

File: projects/core/components/hints-host/hint-box/hint-box.component.ts (hunk 3 above)
Comment (splincode) [reply projects/core/components/hints-host/hint-box/hint-box.component.ts]: fixed

File: projects/core/abstract/abstract-hint-options.ts (hunk 1 above)
Comment (splincode) [reply projects/core/abstract/abstract-hint-options.ts]: fixed

File: projects/addon-charts/components/axes/axes.component.ts [hunk 17]
```diff
@@ -42,7 +42,7 @@ export class TuiAxesComponent {
 
     @Input()
     @tuiDefaultProp()
-    axisYLabels: readonly string[] = [];
+    axisYLabels: readonly (string | null)[] = [];
```
Comment (waterplea) [inline projects/addon-charts/components/axes/axes.component.ts:45]: I believe only `axisXLabels` has special case for `null` to hide ticks, all other arrays don't care for `null`.

File: projects/demo/src/modules/components/carousel/examples/4/index.html (hunk 9 above)
Comment (waterplea) [reply projects/demo/src/modules/components/carousel/examples/4/index.html:10]: Sure, but this seems important, let's not forget about it.

File: projects/kit/components/files/file/file.template.html [hunk 18]
```diff
@@ -52,7 +52,7 @@
         icon="tuiIconCloseLarge"
         appearance="icon"
         class="t-remove"
-        [title]="texts.remove"
+        [title]="$any(texts).remove"
```
Comment (waterplea) [inline projects/kit/components/files/file/file.template.html:55]: Why?

File: projects/kit/components/input-count/input-count.template.html [hunk 19]
```diff
@@ -35,7 +35,7 @@
             class="button button_plus"
             [icon]="iconUp"
             [appearance]="options.appearance"
-            [title]="texts[0]"
+            [title]="$any(texts)[0]"
```
Comment (waterplea) [inline projects/kit/components/input-count/input-count.template.html:38]: Why again and below?

File: projects/kit/components/input-date-time/input-date-time.template.html [hunk 20]
```diff
@@ -1,7 +1,7 @@
 <tui-hosted-dropdown
     class="hosted"
     [canOpen]="interactive"
-    [content]="dropdown"
+    [content]="$explicit(dropdown)"
```
Comment (waterplea) [inline projects/kit/components/input-date-time/input-date-time.template.html:4]: You can pass type to `polymorpheus` instead:
```html
<ng-template #dropdown="polymorpheus" [polymorpheus]="type">
```
and
```ts
readonly type!: TuiContextWithImplicit<TuiActiveZoneDirective>;
```

File: projects/kit/components/input-date/input-date.component.ts [hunk 21]
```diff
@@ -207,6 +212,10 @@ export class TuiInputDateComponent
         }
     }
 
+    $explicit(content: PolymorpheusContent) {
```
Comment (waterplea) [inline projects/kit/components/input-date/input-date.component.ts:215]: Same here

File: projects/kit/components/input-file/file/file.template.html [hunk 22]
```diff
@@ -62,15 +62,15 @@
...
+        *ngIf="(allowDelete && (fileTexts$ | async)) as texts"
         automation-id="tui-file__remove"
         tuiIconButton
         type="button"
         size="xs"
         icon="tuiIconCloseLarge"
         appearance="icon"
         class="remove"
-        [title]="texts.remove"
+        [title]="$any(texts).remove"
```
Comment (waterplea) [inline projects/kit/components/input-file/file/file.template.html:73]: Why is this needed?

File: projects/kit/components/input-file/input-file.template.html [hunk 23]
```diff
@@ -39,11 +39,11 @@
             [accept]="accept"
             [multiple]="multiple"
             [tuiFocusable]="focusable"
-            (change)="onFilesSelected(input, texts, units)"
+            (change)="onFilesSelected(input, $any(texts), units)"
```
Comment (waterplea) [inline projects/kit/components/input-file/input-file.template.html:42]: Again, what's the issue with translations type?

File: projects/kit/components/input-files/input-files.template.html [hunk 24]
```diff
@@ -35,11 +35,11 @@
             [accept]="accept"
             [multiple]="multiple"
             [tuiFocusable]="focusable"
-            (change)="onFilesSelected(input, texts, units)"
+            (change)="onFilesSelected(input, $any(texts), units)"
```
Comment (waterplea) [inline projects/kit/components/input-files/input-files.template.html:38]: Again

File: projects/kit/components/input-password/input-password.template.html [hunk 25]
```diff
@@ -9,7 +9,7 @@
     [tuiTextfieldType]="inputType"
     [disabled]="computedDisabled"
     [readOnly]="readOnly"
-    [iconContent]="interactive ? iconContent : null"
+    [iconContent]="interactive ? $explicit(iconContent) : null"
```
Comment (waterplea) [inline projects/kit/components/input-password/input-password.template.html:12]: Same here

File: projects/kit/components/input-phone/input-phone.component.ts [hunk 26]
```diff
@@ -196,7 +197,9 @@ export class TuiInputPhoneComponent
         }
     }
 
-    onBackspace(event: Event & {target: HTMLInputElement}) {
+    onBackspace(keyboardEvent: KeyboardEvent) {
+        const event = keyboardEvent as unknown as Event & {target: HTMLInputElement};
```
Comment (waterplea) [inline projects/kit/components/input-phone/input-phone.component.ts:201]: I think it's better to typecast target here.

File: projects/kit/components/input-tag/input-tag.component.ts [hunk 27]
```diff
@@ -85,7 +86,7 @@ const EVENT_Y_TO_X_COEFFICIENT = 3;
         HINT_CONTROLLER_PROVIDER,
     ],
 })
-export class TuiInputTagComponent
+export class TuiInputTagComponent<T = string>
```
Comment (waterplea) [inline projects/kit/components/input-tag/input-tag.component.ts:89]: This would break build for everybody who use this class in code. I think we can just use `any` in `disabledItemHandler`.

File: projects/kit/components/pagination/pagination.template.html [hunk 28]
```diff
@@ -8,7 +10,7 @@
             appearance="flat"
             icon="tuiIconChevronLeft"
             class="button"
-            [title]="texts[0]"
+            [title]="$any(texts)[0]"
```
Comment (waterplea) [inline projects/kit/components/pagination/pagination.template.html:13]: And here

File: projects/kit/components/input-count/input-count.template.html (hunk 19 above)
Comment (splincode) [reply projects/kit/components/input-count/input-count.template.html:38]: image

because `text` type is `[string, string] | null`

File: projects/kit/components/input-file/file/file.template.html (hunk 22 above)
Comment (splincode) [reply projects/kit/components/input-file/file/file.template.html:73]: image

File: projects/kit/components/input-file/input-file.template.html (hunk 23 above)
Comment (splincode) [reply projects/kit/components/input-file/input-file.template.html:42]: image

File: projects/kit/components/input-files/input-files.template.html (hunk 24 above)
Comment (splincode) [reply projects/kit/components/input-files/input-files.template.html:38]: image

File: projects/kit/components/pagination/pagination.template.html (hunk 28 above)
Comment (splincode) [reply projects/kit/components/pagination/pagination.template.html:13]: because `text` type is `[string, string] | null`

File: projects/kit/components/files/file/file.template.html (hunk 18 above)
Comment (splincode) [reply projects/kit/components/files/file/file.template.html:55]: image

File: projects/demo/src/modules/components/input-files/examples/3/index.ts [hunk 29]
```diff
@@ -48,7 +48,11 @@ export class TuiInputFilesExample3 {
         this.loadingFile = null;
     }
 
-    restore(file: TuiFileLike) {
+    restore(file: TuiFileLike | null) {
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/input-files/examples/3/index.ts:51]: image
image

`file` can be null

File: projects/cdk/directives/for/for.directive.ts [hunk 30]
```diff
@@ -10,19 +11,19 @@ import {
 @Directive({
     selector: '[ngFor][ngForOf][ngForElse],[ngFor][ngForOf][ngForEmpty]',
 })
-export class TuiForDirective implements OnChanges {
+export class TuiForDirective<T, K = {}> implements OnChanges {
     private ref?: EmbeddedViewRef<{}>;
 
     @Input()
-    ngForOf: unknown[] | null = [];
+    ngForOf: T[] | readonly T[] | null = [];
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/directives/for/for.directive.ts:18]: image

[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/directives/for/for.directive.ts:18]: If we use `T[] | readonly T[] | null` then 

`<div *ngFor="let item of items$ | async else loading empty blank">{{item}}</div>`

`item` is string

If we use `unknown[] | null` then 

`<div *ngFor="let item of items$ | async else loading empty blank">{{item}}</div>`

`item` is unknown

File: projects/addon-table/components/table/th-group/th-group.component.ts [hunk 31]
```diff
@@ -42,4 +42,8 @@ export class TuiThGroupComponent<T> {
         @Inject(forwardRef(() => TuiTableDirective))
         readonly table: TuiTableDirective<T>,
     ) {}
+
+    $string(key: keyof T | string) {
+        return key as string;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-table/components/table/th-group/th-group.component.ts:47]: image

File: projects/addon-charts/components/axes/axes.component.ts (hunk 17 above)
Comment (splincode) [reply projects/addon-charts/components/axes/axes.component.ts]: fixed

File: projects/addon-charts/components/line-chart/line-chart.template.html [hunk 32]
```diff
@@ -72,10 +72,11 @@
                 [style.bottom.%]="getBottom(point[1])"
             ></div>
             <ng-template #hint>
+                <!-- TODO: remove $any after upgrade angular to 10+ -->
                 <div
                     polymorpheus-outlet
                     class="t-text"
-                    [content]="computedHint"
+                    [content]="$any(computedHint)"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-charts/components/line-chart/line-chart.template.html:79]: I don't know how fix it

File: projects/addon-table/components/table/th-group/th-group.component.ts (hunk 31 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-table/components/table/th-group/th-group.component.ts]: fixed

File: projects/core/components/hosted-dropdown/hosted-dropdown.component.ts [hunk 33]
```diff
@@ -60,7 +60,7 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
 
     @Input()
     @tuiDefaultProp()
-    content: PolymorpheusContent = '';
+    content: PolymorpheusContent<{$implicit: TuiActiveZoneDirective}> = '';
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/hosted-dropdown/hosted-dropdown.component.ts]: fixed

File: projects/core/components/hosted-dropdown/hosted-dropdown.component.ts [hunk 34]
```diff
@@ -77,7 +77,7 @@ export class TuiHostedDropdownComponent implements TuiFocusableElementAccessor {
     readonly focusedChange = new EventEmitter<boolean>();
 
     @ViewChild(TuiActiveZoneDirective)
-    readonly activeZone?: TuiActiveZoneDirective;
+    readonly activeZone!: TuiActiveZoneDirective;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/hosted-dropdown/hosted-dropdown.component.ts:81]: fixed

File: projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts [hunk 35]
```diff
@@ -63,7 +63,7 @@ export class ExampleTuiLineDaysChartComponent {
             [new TuiDay(2020, 1, 25), 150],
             [new TuiDay(2020, 1, 27), 10],
         ],
-    ];
+    ] as ReadonlyArray<ReadonlyArray<[TuiDay, number]>>;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/charts/line-days-chart/line-days-chart.component.ts]: fixed

File: projects/demo/src/modules/components/input-files/examples/3/index.ts (hunk 29 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/input-files/examples/3/index.ts:51]: fixed

File: projects/kit/components/input-count/input-count.template.html (hunk 19 above)
Comment (splincode) [reply projects/kit/components/input-count/input-count.template.html:38]: fixed

File: projects/kit/components/input-date-time/input-date-time.template.html (hunk 20 above)
Comment (splincode) [reply projects/kit/components/input-date-time/input-date-time.template.html]: fixed

File: projects/kit/components/input-date/input-date.component.ts (hunk 21 above)
Comment (splincode) [reply projects/kit/components/input-date/input-date.component.ts]: fixed

File: projects/kit/components/input-file/file/file.template.html (hunk 22 above)
Comment (splincode) [reply projects/kit/components/input-file/file/file.template.html:73]: fixed

File: projects/kit/components/input-file/input-file.template.html (hunk 23 above)
Comment (splincode) [reply projects/kit/components/input-file/input-file.template.html]: fixed

File: projects/kit/components/input-password/input-password.template.html (hunk 25 above)
Comment (splincode) [reply projects/kit/components/input-password/input-password.template.html]: fixed

File: projects/kit/components/input-files/input-files.template.html (hunk 24 above)
Comment (splincode) [reply projects/kit/components/input-files/input-files.template.html]: fixed

File: projects/kit/components/files/file/file.template.html (hunk 18 above)
Comment (splincode) [reply projects/kit/components/files/file/file.template.html]: fixed

File: projects/cdk/interfaces/focusable-element-accessor.ts [hunk 36]
```diff
@@ -1,6 +1,8 @@
 import {Observable} from 'rxjs';
 
-export interface TuiNativeFocusableElement extends Element, HTMLOrSVGElement {}
+interface TuiNativeElement extends Element, HTMLOrSVGElement {}
+
+export type TuiNativeFocusableElement = TuiNativeElement & HTMLElement;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/interfaces/focusable-element-accessor.ts]: fixed

File: projects/kit/components/input-phone/input-phone.component.ts (hunk 26 above)
Comment (splincode) [reply projects/kit/components/input-phone/input-phone.component.ts]: fixed

File: projects/kit/components/input-tag/input-tag.component.ts (hunk 27 above)
Comment (splincode) [reply projects/kit/components/input-tag/input-tag.component.ts]: fixed

File: projects/kit/components/pagination/pagination.template.html (hunk 28 above)
Comment (splincode) [reply projects/kit/components/pagination/pagination.template.html]: fixed

File: projects/kit/components/input-date/input-date.component.ts [hunk 37]
```diff
@@ -190,8 +196,10 @@ export class TuiInputDateComponent
         return this.interactive && !this.computedMobile;
     }
 
-    get computedMask(): TuiTextMaskOptions {
-        return this.activeItem ? EMPTY_MASK : this.textMaskOptions;
+    get computedMask(): TextMaskConfig {
+        return (this.activeItem
+            ? EMPTY_MASK
+            : this.textMaskOptions) as TuiTextMaskOptions as unknown as TextMaskConfig;
```
Comment (waterplea) [inline projects/kit/components/input-date/input-date.component.ts:202]: Why don't you type `textMaskOptions` at the declaration instead?

File: projects/kit/components/input-date/input-date.template.html [hunk 38]
```diff
@@ -46,7 +46,7 @@
         ></tui-svg>
     </ng-template>
 
-    <ng-template #dropdown="polymorpheus" polymorpheus>
+    <ng-template #dropdown [polymorpheus]="type">
```
Comment (waterplea) [inline projects/kit/components/input-date/input-date.template.html:49]: Same here

File: projects/kit/components/input-password/input-password.template.html [hunk 39]
```diff
@@ -20,7 +20,7 @@
     <ng-content select="input" ngProjectAs="input"></ng-content>
 </tui-primitive-textfield>
 
-<ng-template #iconContent="polymorpheus" polymorpheus>
+<ng-template #iconContent [polymorpheus]="type">
```
Comment (waterplea) [inline projects/kit/components/input-password/input-password.template.html:23]: And here

File: projects/addon-charts/components/line-chart/line-chart.template.html (hunk 32 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-charts/components/line-chart/line-chart.template.html:79]: image

`Type 'TuiContextWithImplicit<readonly (readonly [number, number])[]>' is not assignable to type 'TuiContextWithImplicit<readonly [number, number]>'`

I tried to fix it but it doesn't work

File: projects/kit/components/input-date-time/input-date-time.template.html [hunk 40]
```diff
@@ -35,7 +35,7 @@
         ></tui-svg>
     </ng-template>
 
-    <ng-template #dropdown="polymorpheus" polymorpheus>
+    <ng-template #dropdown [polymorpheus]="type">
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-date-time/input-date-time.template.html:38]: image
image

I tried to do this, but for some reason the type does not match then

File: projects/kit/components/input-date/input-date.component.ts (hunk 37 above)
Comment (splincode) [reply projects/kit/components/input-date/input-date.component.ts:202]: then there will be problems

image

image

File: projects/kit/components/input-date/input-date.template.html (hunk 38 above)
Comment (splincode) [reply projects/kit/components/input-date/input-date.template.html]: fixed

File: projects/kit/components/input-password/input-password.template.html (hunk 39 above)
Comment (splincode) [reply projects/kit/components/input-password/input-password.template.html]: fixed

File: projects/kit/components/input-date-time/input-date-time.template.html (hunk 40 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/kit/components/input-date-time/input-date-time.template.html]: fixed

File: projects/kit/components/input-date/input-date.component.ts (hunk 37 above)
Comment (waterplea) [reply projects/kit/components/input-date/input-date.component.ts:202]: You type getter as `TextMaskConfig` and typecast `this.textMaskOptions` at the place where it's defined. It's a readonly prop, let's even move it to a constant so the name is minified.

Comment (splincode) [reply projects/kit/components/input-date/input-date.component.ts:202]: Sorry, I don't understand (( I can't do different manipulations due to type mismatch

File: projects/addon-charts/components/line-chart/line-chart.template.html (hunk 32 above)
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/addon-charts/components/line-chart/line-chart.template.html:79]: I've fixed it.

**Review by waterplea — APPROVED** [review]

File: projects/addon-charts/components/line-chart/line-chart.template.html (hunk 32 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-charts/components/line-chart/line-chart.template.html]: Thank you very much  🙏
